### PR TITLE
Add ability for verbosity levels in installation

### PR DESCRIPTION
**Pull request by mikeymike**
@AydinHassan this allows for `-vvv` to be passed through to composer during installation. Couldn't really find a better way to do this, let me know what you think :+1:

### CODE REVIEW

**Comment by AydinHassan** [comment]
So does this allow you to install a workshop like `workshop-manager install some-workshop -vvv` and see the composer output?

**Comment by mikeymike** [comment]
Yeah man exactly that 😎

File: src/Installer.php [hunk 1]
```diff
@@ -37,41 +37,44 @@
...
 
     /**
      * @param Workshop $workshop
-     * @throws WorkshopAlreadyInstalledException
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @throws \League\Flysystem\FileExistsException
      */
-    public function installWorkshop(Workshop $workshop)
+    public function installWorkshop(Workshop $workshop, InputInterface $input, OutputInterface $output)
```
Comment (AydinHassan) [inline src/Installer.php]: I think maybe the factory should be in the command so this class doesn't depend on the Symfony `InputInterface` and `OutputInterface`

File: app/config.php [hunk 2]
```diff
@@ -75,6 +76,7 @@
     }),
     Client::class => \DI\object(Client::class),
     Factory::class => \DI\object(),
+    IOFactory::class => \DI\object(IOFactory::class),
```
Comment (AydinHassan) [inline app/config.php]: You can omit the argument to `object` it is inferred from the registry key

File: src/IOFactory.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     * @return ConsoleIO|NullIO
+     */
+    public function getIO(InputInterface $input, OutputInterface $output)
+    {
+        switch ($output->getVerbosity()) {
+            case OutputInterface::VERBOSITY_VERBOSE:
+            case OutputInterface::VERBOSITY_VERY_VERBOSE:
+            case OutputInterface::VERBOSITY_DEBUG:
+                return new ConsoleIO($input, $output, new HelperSet);
+                break;
```
Comment (AydinHassan) [inline src/IOFactory.php]: these `break`'s are useless :smile:

File: src/Installer.php (hunk 1 above)
Comment (mikeymike) [inline src/Installer.php]: I won't be able to do the DI on the installer though will I ?

File: src/IOFactory.php (hunk 3 above)
Comment (mikeymike) [inline src/IOFactory.php]: :joy: force of habit, I actually put the breaks in before I put the code in

File: src/Installer.php (hunk 1 above)
Comment (mikeymike) [inline src/Installer.php]: Ignore that just me being a turd.. I get what you mean :+1:

**Pull request merged**